chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiDataRetriever.java

```diff
@@ -0,0 +1,267 @@
+package org.reactome.release.chebiupdate;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.schema.InvalidAttributeException;
+
+import uk.ac.ebi.chebi.webapps.chebiWS.client.ChebiWebServiceClient;
+import uk.ac.ebi.chebi.webapps.chebiWS.model.ChebiWebServiceFault_Exception;
+import uk.ac.ebi.chebi.webapps.chebiWS.model.DataItem;
+import uk.ac.ebi.chebi.webapps.chebiWS.model.Entity;
+
+/**
+ * An instance of this class can be used to retrieve data from ChEBI.
+ * @author sshorser
+ *
+ */
+class ChebiDataRetriever
+{
+	private ChebiWebServiceClient chebiClient = new ChebiWebServiceClient();
+	
+	private static final Logger logger = LogManager.getLogger();
+	
+	private boolean useCache;
+
+	/**
+	 * Creates a new ChEBI Data Retriever. 
+	 * @param useCache - should the cache be used? If <em>true</em> the file "chebi-cache" will be used and no live connections to ChEBI will be attempted.
+	 */
+	public ChebiDataRetriever(boolean useCache)
+	{
+		this.useCache = useCache;
+	}
+	
+	/**
+	 * This looks weird, I know. I needed to be able to set the Formulae on an Entity 
+	 * and the Entity class provided by ChEBI does not have a setter for that. It has setters for
+	 * other members, just not all of them. So I added a setter, hence the "accessible" name.
+	 * @author sshorser
+	 *
+	 */
+	class AccessibleEntity extends Entity
+	{
+		public void setFormulae(List<DataItem> formulae)
+		{
+			this.formulae = formulae;
+		}
+	}
+	
+	
+	/**
+	 * Makes calls to the ChEBI web service to get info for specified ChEBI
+	 * identifiers.
+	 * 
+	 * @param updator
+	 * @param refMolecules - a list of ReferenceMolecules. The Identifier of each of these molecules will be sent to ChEBI to get up-to-date information for that Identifier.
+	 * @param failedEntitiesList - A list of ReferenceMolecules for which no information was returned by ChEBI. Will be updated by this method.
+	 * @return A ReferenceMolecule DB_ID-to-ChEBI Entity map.
+	 * @throws IOException 
+	 */
+	public Map<Long, Entity> retrieveUpdatesFromChebi(Collection<GKInstance> refMolecules, Map<GKInstance, String> failedEntitiesList) throws IOException
```

Comment:
What is this Entity type?

Reply:
`uk.ac.ebi.chebi.webapps.chebiWS.model.Entity` comes from the ChEBI WebService Client dependency. It's a part of their data model for ChEBI data.